tests/test_assembly.py

```diff
@@ -279,6 +593,231 @@ def test_save_raises(nested_assy):
+                (
+                    ["chassis", "wheel-axle.*", "axle"],
+                    {
+                        "color": (0.0, 1.0, 0.0, 1.0),
+                        "color_shape": (0.0, 1.0, 0.0, 1.0),
+                        "num_nodes": 2,
+                    },
+                ),
+                (
+                    ["chassis", "wheel-axle.*", "axle", "axle_part"],
+                    {"color": (0.0, 1.0, 0.0, 1.0), "num_nodes": 2},
+                ),
+            ],
+        ),
+    ],
+)
+def test_colors_assy0(assy_fixture, expected, request):
+    """Validate assembly colors with document explorer.
+
+    Check toCAF wth color shape parameter False.
+    """
+
+    def check_nodes(doc, expected):
+        allnodes = get_doc_nodes(doc, False)
+        for name_path, props in expected:
+            nodes = find_node(allnodes, name_path)
+            if "num_nodes" in props:
+                assert len(nodes) == props["num_nodes"]
+                props.pop("num_nodes")
+            else:
+                assert len(nodes) > 0
+            for n in nodes:
+                for k, v in props.items():
+                    assert pytest.approx(n[k], abs=1e-3) == v
+
+    assy = request.getfixturevalue(assy_fixture)
+    _, doc = toCAF(assy, False)
+    check_nodes(doc, expected)
+
+
+@pytest.mark.parametrize(
+    "assy_fixture, expected",
+    [
+        (
+            "nested_assy",
+            [
+                (
+                    ["TOP", "SECOND", "SECOND_part"],
+                    {"color_shape": (0.0, 1.0, 0.0, 1.0)},
+                ),
+                (
+                    ["TOP", "SECOND", "BOTTOM", "BOTTOM_part"],
+                    {
+                        "color_shape": (0.0, 1.0, 0.0, 1.0),
+                        "color_subshapes": (0.0, 1.0, 0.0, 1.0),
+                    },
+                ),
+            ],
+        ),
+        ("empty_top_assy", [([".*_part"], {"color_shape": (0.0, 1.0, 0.0, 1.0)}),]),
+        (
+            "boxes0_assy",
+            [
+                (["box0", "box0_part"], {"color_shape": (1.0, 0.0, 0.0, 1.0)}),
+                (["box1", "box1_part"], {"color_shape": (1.0, 0.0, 0.0, 1.0)}),
+            ],
+        ),
+        (
+            "boxes1_assy",
+            [
+                (["box0", "box0_part"], {"color_shape": (1.0, 0.0, 0.0, 1.0)}),
+                (["box1", "box0_part"], {"color_shape": (1.0, 0.0, 0.0, 1.0)}),
+            ],
+        ),
+        (
+            "boxes2_assy",
+            [
+                (["box0", "box0_part"], {"color_shape": (1.0, 0.0, 0.0, 1.0)}),
+                (["box1", "box1_part"], {"color_shape": (0.0, 1.0, 0.0, 1.0)}),
+            ],
+        ),
+        (
+            "boxes3_assy",
+            [
+                (["box0", "box0_part"], {"color_shape": (1.0, 0.0, 0.0, 1.0)}),
+                (
+                    ["box1", "box1_part"],
+                    {"color_shape": cq.Color().toTuple()},
```

Comment:
Change line to following with revert of Color changes (disallow default color e.g cq.Color()):
```py
{"color_shape": (*Quantity_ColorRGBA().GetRGB().Values(Quantity_TOC_RGB), Quantity_ColorRGBA().Alpha()) },
```

or perhaps (as default Alpha unlikely to change in a future OCCT update):


```py
{"color_shape": (*Quantity_ColorRGBA().GetRGB().Values(Quantity_TOC_RGB), 1.0)},
```